pyomo/contrib/doe/fim_doe.py

```diff
@@ -0,0 +1,2582 @@
+            if True, the parameters are scaled by its own nominal value in param_init
+        scale_constant_value:
+            scale all elements in Jacobian matrix, default is 1.
+        store_name:
+            a string of file name. If not None, store results with this name.
+            Since there are maultiple experiments, results are numbered with a scalar number, 
+            and the result for one grid is 'store_name(count).csv' (count is the number of count).
+        read_name: 
+            a string of file name. If not None, read result files. 
+            Since there are multiple experiments, this string should be the common part of all files.
+        filename:
+            if True, grid search results stored with this file name
+        formula:
+            choose from 'central', 'forward', 'backward', None. This option is only used for 'sequential_finite' mode.
+        step:
+            Sensitivity perturbation step size, a fraction between [0,1]. default is 0.001
+
+        Return
+        -------
+        figure_draw_object: a combined result object of class Grid_search_result
+        """
+
+        # time 0
+        t_enumeration_begin = time.time()
+
+        # Set the Objective Function to 0 helps solve square problem quickly
+        self.objective_option='zero'
+        self.filename = filename
+
+        # calculate how much the FIM element is scaled
+        self.fim_scale_constant_value = scale_constant_value ** 2
+
+        # when defining design space, design variable values are defined as in design_values argument
+        # the design var value defined in dv_ranges only applies to control time points given in dv_apply_time
+        grid_dimension = len(design_ranges)
+
+        # to store all FIM results
+        result_combine = {}
+
+        # iteration 0
+        count = 0
+        failed_count = 0
+        # how many sets of design variables will be run
+        total_count = 1
+        for i in range(grid_dimension):
+            total_count *= len(design_ranges[i])
+
+        # generate combinations of design variable values to go over
+        search_design_set = product(*design_ranges)
+
+        build_time_store=[]
+        solve_time_store=[]
+
+        # loop over design value combinations
+        for design_set_iter in search_design_set:
+            # generate the design variable dictionary needed for running compute_FIM
+            # first copy value from design_Values
+            design_iter = design_values.copy()
+
+            # update the controlled value of certain time points for certain design variables
+            for i in range(grid_dimension):
+                for v, value in enumerate(design_control_time[i]):
+                    design_iter[design_dimension_names[i]][value] = list(design_set_iter)[i]
+
+            self.logger.info('=======This is the ', count+1, 'th iteration=======')
+            self.logger.debug('Design variable values of this iteration:', design_iter)
```

Comment:
Again, you are misunderstanding the logging API: it is not the same as `print()`: the first argument is the formatting string and the remaining positional arguments are merged with it using the string formatting operator.  You want to change the first info() call to:
```python
self.logger.info('=======This is the %sth iteration=======', count+1)
```